### PR TITLE
Feat/completion

### PR DESCRIPTION
**Pull request by GabrielBrandao1618**
So, the completion command works the following way:

`quartz completion --shell <shell>`

The only thing I am kind of struggling to implement is this detailed description that github CLI provides. I couldn't find a way to implement this `flags` section specifying the available shells.
image

### CODE REVIEW

File: src/action/completion.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        _ => None,
+    }
+}
+
+pub fn completion(args: CompletionArgs) {
+    if let Some(shell) = get_shell_by_name(&args.shell) {
+        let mut cmd = get_command();
+        print_completions(shell, &mut cmd);
+    } else {
+        eprintln!("No such shell was found: {}", &args.shell);
```
Comment (EduardoRodriguesF) [inline src/action/completion.rs:35]: Make it more consistent with similar errors like `no such handle`.

```suggestion
        eprintln!("no such shell: {}", &args.shell);
```

File: src/action/completion.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        "zsh" => Some(Shell::Zsh),
+        "bash" => Some(Shell::Bash),
+        "fish" => Some(Shell::Fish),
+        "powershell" => Some(Shell::PowerShell),
+        "elvish" => Some(Shell::Elvish),
+        _ => None,
+    }
+}
+
+pub fn completion(args: CompletionArgs) {
```
Comment (EduardoRodriguesF) [inline src/action/completion.rs:30]: Probably my bad for not writing about coding styles, but the pattern uses `cmd` for the main action function. That way, so we can call `action::completion::cmd`.

```suggestion
pub fn cmd(args: CompletionArgs) {
```

**Review by EduardoRodriguesF — CHANGES_REQUESTED** [review]

**Comment by EduardoRodriguesF** [comment]
> I couldn't find a way to implement this flags section specifying the available shells.
We can be more practical and change how it displays the value, altering the `value_name` attribute to `bash|zsh|...`.

File: src/cli.rs [hunk 3]
```diff
@@ -92,6 +92,11 @@ pub enum Cmd {
         #[command(subcommand)]
         command: ConfigCmd,
     },
+    #[command(
+        name = "completion",
+        about = "outputs a completion script for a given shell"
+    )]
```
Comment (EduardoRodriguesF) [inline src/cli.rs:98]: First letter should be capitalize. Also, let's stick with description through comment to keep the file consistent. Change "outputs" to "print",  to also keep consistency.
```suggestion
    // Print completion script for a given shell
    #[command(name = "completion")]
```

**Review by EduardoRodriguesF — CHANGES_REQUESTED** [review]

**Comment by EduardoRodriguesF** [comment]
Hey, I just found out that there is a way to define enums as arguments.

https://www.rustadventure.dev/introducing-clap/clap-v4/introducing-enums-as-clap-arguments

Could you implement that for the shell flag?

File: src/action/completion.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+fn print_completions<G: Generator>(gen: G, cmd: &mut Command) {
+    generate(gen, cmd, cmd.get_name().to_string(), &mut std::io::stdout());
+}
+
+fn get_shell_by_name(shell_name: &str) -> Option<Shell> {
+    match shell_name {
+        "zsh" => Some(Shell::Zsh),
+        "bash" => Some(Shell::Bash),
+        "fish" => Some(Shell::Fish),
+        "powershell" => Some(Shell::PowerShell),
```
Comment (EduardoRodriguesF) [inline src/action/completion.rs:24]: We should remove powershell completion, since quartz does not work on Windows yet.
```suggestion
```

**Review by EduardoRodriguesF — CHANGES_REQUESTED** [review]

**Comment by GabrielBrandao1618** [comment]
> Hey, I just found out that there is a way to define enums as arguments.
> 
> https://www.rustadventure.dev/introducing-clap/clap-v4/introducing-enums-as-clap-arguments
> 
> Could you implement that for the shell flag?

Oh, that's great! I am definitively checking this out ASAP

Man, I wish Clap docs had more examples since it relies a lot in macros😓

**Comment by EduardoRodriguesF** [comment]
>  Man, I wish Clap docs had more examples since it relies a lot in macros😓

Yeah, their docs on macros are pretty scarce. I was considering refactoring it to the builder format, but it would take some time to do so.

**Review by EduardoRodriguesF — APPROVED** [review]

**Comment by EduardoRodriguesF** [comment]
Merging on a separate branch so we can work fully on #36 before actually releasing.

**Pull request merged**